### PR TITLE
[v2-beta] Improvement - Editor

### PR DESCRIPTION
**Pull request by gsmithun4**

### CODE REVIEW

File: frontend/src/_components/SearchBox.jsx [hunk 1]
```diff
@@ -22,8 +22,15 @@ export function SearchBox({
...
   };
 
+  const initialRender = useRef(true);
+
   useEffect(() => {
-    onSubmit(debouncedSearchTerm);
+    if (initialRender.current) {
+      // To avoid calling onSubmit on initial load
+      initialRender.current = false;
+    } else {
```
Comment (kavinvenkatachalam) [inline frontend/src/_components/SearchBox.jsx:31]: `useMounted` hook can be used here.

File: frontend/src/AppLoader/AppLoader.jsx [hunk 2]
```diff
@@ -198,7 +198,13 @@ const AppLoaderComponent = (props) => {
       ) : (
         <>
           {appDetails ? (
-            <>{config.ENABLE_MULTIPLAYER_EDITING ? <RealtimeEditor {...props} /> : <Editor {...props} />} </>
+            <>
+              {config.ENABLE_MULTIPLAYER_EDITING ? (
+                <RealtimeEditor {...props} appDetails={appDetails} />
+              ) : (
+                <Editor {...props} appDetails={appDetails} />
+              )}{' '}
```
Comment (kavinvenkatachalam) [inline frontend/src/AppLoader/AppLoader.jsx:206]:
```suggestion
              )}
```

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@gsmithun4 Suggested a minor changes

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Hey @gsmithun4 , when a version is released, the app automatically switches to the home page instead of staying in the current page. Please check.

**Review by sherfin94 — APPROVED** [review]

**Review by kavinvenkatachalam — APPROVED** [review]
LGTM

**Pull request merged**